### PR TITLE
deps: upgrade aegir to 40.x.x

### PR DESCRIPTION
**Pull request by achingbrain**
Upgrades aegir and fixes linting errors and typedoc config.

### CODE REVIEW

File: packages/interface-internal/package.json [hunk 1]
```diff
@@ -60,10 +60,6 @@
       "types": "./dist/src/registrar/index.d.ts",
       "import": "./dist/src/registrar/index.js"
     },
-    "./transport": {
```
Comment (maschad) [inline packages/interface-internal/package.json:63]: Why are these types no longer exported?

File: interop/BrowserDockerfile [hunk 2]
```diff
@@ -4,10 +4,11 @@ FROM mcr.microsoft.com/playwright
 
 COPY --from=node-js-libp2p-head /app/ /app/
 WORKDIR /app/interop
-RUN npx playwright install
```
Comment (maschad) [inline interop/BrowserDockerfile:7]: Why is it we no longer need to install the playwright deps?

Comment (achingbrain) [reply interop/BrowserDockerfile:7]: playwright-test does it automatically if needed

File: packages/interface-internal/package.json (hunk 1 above)
Comment (achingbrain) [reply packages/interface-internal/package.json:63]: These files don’t exist in this module

**Pull request merged**